src/Datasource/EntityTrait.php

```diff
@@ -283,6 +320,23 @@ public function &get($property)
             throw new InvalidArgumentException('Cannot get an empty property');
         }
 
+        $nested = false;
+        if (strpos($property, '.') > 0 && !array_key_exists($property, $this->_properties)) {
+            list($property, $nested) = explode('.', $property, 2);
+        }
+
+        if ($nested) {
+            if ($this->_properties[$property] instanceof EntityInterface) {
+                $result = $this->_properties[$property]->get($nested);
+                return $result;
+            } elseif (is_array($this->_properties[$property])
+                || ($this->_properties[$property] instanceof \ArrayAccess)
+            ) {
+                $value = Hash::get($this->_properties[$property], $nested);
+                return $value; //@TODO: Or remove this line and let getters engage below?
```

Comment:
The argument for doing that is: "the last applicable getter should fire."

Consider:

```php
$entity = new Entity([
    'a' => [
        'b' => 'value'
    ],
]);
echo $entity->a['b']; // Engages `_getA()`, if it exists.
echo $entity->get('a.b'); // Should probably still call `_getA()` before using `Hash::get('b')`??
```